api/v1alpha1/akodeploymentconfig_webhook.go

```diff
@@ -145,22 +145,39 @@ func (r *AKODeploymentConfig) validateAVI(old *AKODeploymentConfig) field.ErrorL
 	}
 
 	// check avi controller version format
-	controllerVersion, err := r.validateAviControllerVersion()
+	_, err := r.validateAviControllerVersion()
 	if err != nil {
 		allErrs = append(allErrs, err)
 	}
 
 	if !runTest {
-		client, err := aviclient.NewAviClient(&aviclient.AviClientConfig{
-			ServerIP: r.Spec.Controller,
-			Username: string(adminCredential.Data["username"][:]),
-			Password: string(adminCredential.Data["password"][:]),
-			CA:       string(aviControllerCA.Data["certificateAuthorityData"][:]),
-		}, controllerVersion)
+		username := string(adminCredential.Data["username"][:])
+		password := string(adminCredential.Data["password"][:])
+		certificate := string(aviControllerCA.Data["certificateAuthorityData"][:])
+
+		client, fieldErr := r.validateAviAccount(username, password, certificate, "")
```

Comment:
what will happen if we are passing empty version `""` to this function and create a new Avi client?

Reply:
It will authorize us to call API to get the correct avi controller version, but we can't use it to call API to get/modify avi resources until we re-init the avi client with the correct version